core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -726,10 +726,11 @@ class HttpGoogleServicesDAO(
             case None => ""
           }))
 
-          val result = executeGoogleRequest(updater)
-
-          span.end()
-          result
+          try {
+            executeGoogleRequest(updater)
+          } finally {
+            span.end()
```

Comment:
From the description, it would seem that every span should be closed in a finally, or there should be some type of transaction/automatic handling of this.

Is there a different syntax that you can do that will automatically close the span and rethrow the exception?

Reply:
Not at the moment; there is something like this in `io.opencensus.scala.Tracing` which we use liberally (`traceWithParent` being most notable), however those are all expecting to wrap async Future's while this is context is not in a Future. If this is a pattern we use again, it'd be useful to extract an automatic resource management method to handle this.